content/posts/2022-02-23-tips-to-succeed-at-a-systems-design-interview/index.md

```diff
@@ -0,0 +1,70 @@
+---
+title: Tips to succeed at a system’s design interview
+author: Luiz Rosa
+twitter: luizguilhermefr
+layout: post
+lang: en
+path: /blog/tips-to-succeed-at-a-systems-design-interview
+date: '2022-02-23'
+comments: true
+---
+
+![cover](./cover.png)
+
+At Loadsmart, the [hiring process](https://github.com/loadsmart/culture/blob/main/hiring/interviewing.md) consists of four steps:
+A screening interview, with one of our recruiters, a coding exercise, done through HackerRank, a system’s design interview, and, finally, a soft-skills interview.
+But what to expect from a system’s design interview, and how to succeed on it?
+First of all, the system’s design, also known as the “architectural” interview has slightly different formats depending on the position you are applying for.
+It will, however, always require you as a candidate to explain how you would approach a given problem, proposing a solution and then iterating over it as more requirements or variables are added.
+It usually comes with a whiteboard so you can visually explain your thoughts and keep track of what you’ve built so far.
+We evaluate not only your technical acumen but also your communication and ability to explain your ideas, according to the seniority of the position you have applied to.
+There are no right or wrong answers, but there are solutions that would work all the time, there are solutions that could work most of the time and there are solutions that would not work at all.
+Here are 5 tips for you to convince the interviewer your solution falls under the first case.
+
+### Tip 1: Pay attention to the problem scope
+
+One of the most common mistakes candidates make on architectural interviews is to go beyond what was asked in the problem statement.
+Imagine that the scope is closed, every requirement is there.
+Later on, more variables and requirements will be added, so we will evaluate your ability to adapt your solution to this new scenario.
+Adding by yourself new requirements to the problem might indicate that you are not comfortable with the actual requirements and are trying to run away from them.
+So the first tip is: Stick to the problem scope.
+
+### Tip 2: Do not over-engineer
+
+We want to evaluate your ability to create complex systems, but also simple ones.
+In a real-world scenario, you would unlikely start a project with dozens of servers and top-notch tools to intercommunicate them.
+Start lean, paying attention to the scope, and then evolve your solution as variables such as requests per minute are added.
+If you’re in doubt, ask the interviewer.
+
+### Tip 3: Understand what you are proposing
+
+In our industry, we are bombarded with new technologies each day, but if you pay attention, just a few of them really lasts.
```

Comment:
It's a great tip! We're seeing several candidates throwing buzzwords trying to impress the interviewers.